### ISSUE TITLE
Allow Razor Components with code behind file to specify base class only once

### ISSUE THREAD

**Issue by Eirenarch** [dotnet/aspnetcore#18750]
I prefer using Razor Components with a code behind .cs file. When specifying a custom base class it is not enough to specify it in the part of the partial class in the .cs file I also need to add @inherits in the .razor file. If I don't do so the compiler complains about different base classes.

I hope it would be possible to only specify the base class in the code behind file and have the tooling detect the presence of a partial file at which point it would not generate a base class from the Razor file. That would result in cleaner .razor files with less directives on top and make them more pure templates which is the whole point of splitting the C# code in a separate file in a separate file.

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us. This is not something we plan to support because this is not a priority.

**Comment by grahamehorner** [comment]
@mkArtakMSFT sorry but closing an issue with a comment "...this is not a priority." is not good, or IMHO the correct action for this; again IMHO this should remain open and be marked as up-for-grabs to allow the community to pick this up and address via a PR; closing the issue means that community may miss this issue and therefore developers that are willing to assist with improvements aren't able to address.
